src/System.Runtime.Caching/tests/System.Runtime.Caching/HostFileChangeMonitorTest.cs

```diff
@@ -0,0 +1,318 @@
+            else
+                missingFile = "/file.txt";
+
+            paths.Clear();
+            paths.Add(missingFile);
+            monitor = new HostFileChangeMonitor(paths);
+            Assert.Equal(1, monitor.FilePaths.Count);
+            Assert.Equal(missingFile, monitor.FilePaths[0]);
+            //??
+            Assert.Equal(missingFile + "701CE1722770000FFFFFFFFFFFFFFFF", monitor.UniqueId);
+            monitor.Dispose();
+
+            paths.Add(missingFile);
+            monitor = new HostFileChangeMonitor(paths);
+            Assert.Equal(2, monitor.FilePaths.Count);
+            Assert.Equal(missingFile, monitor.FilePaths[0]);
+            Assert.Equal(missingFile, monitor.FilePaths[1]);
+            //??
+            Assert.Equal(missingFile + "701CE1722770000FFFFFFFFFFFFFFFF", monitor.UniqueId);
+            monitor.Dispose();
+        }
+
+        [Fact]
+        public void Constructor_Duplicates()
+        {
+            HostFileChangeMonitor monitor;
+            PlatformID pid = Environment.OSVersion.Platform;
+            bool runningOnWindows = ((int)pid != 128 && pid != PlatformID.Unix && pid != PlatformID.MacOSX);
+            string missingFile = Path.Combine("missing", "file", "path");
+
+            if (runningOnWindows)
+                missingFile = "c:\\file.txt";
+            else
+                missingFile = "/file.txt";
+
+            var paths = new List<string> {
+                missingFile,
+                missingFile
+            };
+
+            // Just checks if it doesn't throw any exception for dupes
+            monitor = new HostFileChangeMonitor(paths);
+            monitor.Dispose();
+        }
+
+        private static Tuple<string, string, string, IList<string>> SetupMonitoring()
+        {
+            string testPath = Path.Combine(Path.GetTempPath(), "HostFileChangeMonitorTest", "Dispose_Calls_StopMonitoring");
+            if (!Directory.Exists(testPath))
+                Directory.CreateDirectory(testPath);
+
+            string firstFile = Path.Combine(testPath, "FirstFile.txt");
+            string secondFile = Path.Combine(testPath, "SecondFile.txt");
+
+            File.WriteAllText(firstFile, "I am the first file.");
+            File.WriteAllText(secondFile, "I am the second file.");
+
+            var paths = new List<string> {
+                firstFile,
+                secondFile
+            };
+
+            return new Tuple<string, string, string, IList<string>>(testPath, firstFile, secondFile, paths);
+        }
+
+        private static void CleanupMonitoring(Tuple<string, string, string, IList<string>> setup)
+        {
+            string testPath = setup != null ? setup.Item1 : null;
+            if (String.IsNullOrEmpty(testPath) || !Directory.Exists(testPath))
+                return;
+
+            foreach (string f in Directory.EnumerateFiles(testPath))
+            {
+                try
+                {
+                    File.Delete(f);
```

Comment:
What about the directories that were created?
https://github.com/dotnet/corefx/pull/25169/files#diff-3d01aab996e395051519fea618313e39R169

Reply:
It is always the same folder that the tests use. Therefore the files get created and deleted but the folder stays. Perhaps we should delete the folder as well to stay clean (and perhaps if I were writing those tests from scratch I'd use a new random folder name for each test). 
I see the TempFile & TempDirectory & temp file collection helpers - they aren't really useful here without refactoring these tests.

Currently, because of #25168 there is only one test uses it so I am adding a code to delete the folder. I'll refactor it when fixing #25168.